### PR TITLE
Add an API for scaling down a replicated service by removing a specific replica (task)

### LINKED ISSUE #33426: Proposal: add `--task-rm` option to `service update`
(Related issue #33392)

The replicated orchestrator scales down a service by removing tasks in a way that balances the resulting tasks between nodes. But some stateful services may need to be scaled down by removing specific tasks to minimize the impact of scaling down.

#### Use Case
An elasticsearch cluster can reallocate shards from tasks to be ended to another tasks if it can select tasks for termination when scaling down.

#### Usage
Users can use `--task-rm` flag to scale down a service by removing a specific task.
```
$ docker service update --task-rm [task name] [service name]
```

#### Usage Example
```
$ docker service create --name redis --replicas 3 redis:3.0.6

dmu1ept4cxcfe8k8lhtux3ro3

$ docker service ps redis

ID             NAME      IMAGE        NODE      DESIRED STATE  CURRENT STATE          ERROR  PORTS
0qihejybwf1x   redis.1   redis:3.0.5  manager1  Running        Running 8 seconds
bk658fpbex0d   redis.2   redis:3.0.5  worker2   Running        Running 9 seconds
5ls5s5fldaqg   redis.3   redis:3.0.5  worker1   Running        Running 9 seconds

$ docker service update --task-rm redis.2 redis

redis scaled to 2

$ docker service ps redis

ID             NAME      IMAGE        NODE      DESIRED STATE  CURRENT STATE          ERROR  PORTS
0qihejybwf1x   redis.1   redis:3.0.5  manager1  Running        Running 8 seconds
5ls5s5fldaqg   redis.3   redis:3.0.5  worker1   Running        Running 9 seconds
```

### PR DESCRIPTION
**Pull request by sungwonh**
This PR adds an API for scaling down a replicated service by removing a specific replica (task).
Please refer to #33426 for more details. (closes #33426)

This PR depends on [docker/swarmkit#2209](https://github.com/docker/swarmkit/pull/2209).

### CODE REVIEW

File: api/server/router/swarm/backend.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Backend interface {
 	CreateService(types.ServiceSpec, string, bool) (*basictypes.ServiceCreateResponse, error)
 	UpdateService(string, uint64, types.ServiceSpec, basictypes.ServiceUpdateOptions, bool) (*basictypes.ServiceUpdateResponse, error)
 	RemoveService(string) error
+	RemoveReplica(serviceID string, slot uint64) error
```
Comment (AkihiroSuda) [inline api/server/router/swarm/backend.go]: As @aaronlehmann commented (https://github.com/docker/swarmkit/pull/2209#discussion_r119814950), `RemoveTask` seems better than `RemoveReplica`.

**Comment by thaJeztah** [comment]
I think this PR needs a wider discussion to see if this is the right approach

**Comment by cpuguy83** [comment]
@sungwonh Do you mind if we close this and discuss design on the original issue?

**Comment by sungwonh** [comment]
@cpuguy83 No, I don't mind.

**Comment by cpuguy83** [comment]
Thanks!

**Pull request closed without merging**